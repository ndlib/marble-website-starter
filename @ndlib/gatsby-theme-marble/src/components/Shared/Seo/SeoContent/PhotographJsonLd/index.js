import React from 'react'
import PropTypes from 'prop-types'
import { JsonLd } from '../JsonLd'

export default function PhotographJsonLd ({ title, description, date, image, author, creditText, dimensions, thumbnail }) {
  return (
    <JsonLd>
      {{
        '@context': 'https://schema.org',
        '@type': 'VisualArtwork',
        'name': title,
        'dateCreated': date,
        'description': description,
        'image': image,
        'thumbnailUrl': thumbnail,
        'size': dimensions,
        'artform': 'photograph',
        'creator': [
          {
            '@type': 'Person',
            'name': author,
          }
        ],
        'artMedium': 'photography',
        'creditText': creditText,
      }}
    </JsonLd>
  )
}

PhotographJsonLd.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  creditText: PropTypes.string.isRequired,
  dimensions: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
}
