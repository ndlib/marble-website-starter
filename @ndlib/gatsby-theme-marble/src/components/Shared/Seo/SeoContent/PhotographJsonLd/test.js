import React from 'react'
import renderer from 'react-test-renderer'
import PhotographJsonLd from './'

describe('PhotographJsonLd component', () => {

  it('renders props into values', () => {
    const tree = renderer
      .create(<PhotographJsonLd 
        title={'Fake Title'}
        description={'This is a fake description'}
        image={'http://fake.io/image'}
        author={'Fake Author'}
        creditText={'Fake Creditor'}
        dimensions={'2 x 4'}
        thumbnail={'http://fake.io/thumbnail'}
        />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})