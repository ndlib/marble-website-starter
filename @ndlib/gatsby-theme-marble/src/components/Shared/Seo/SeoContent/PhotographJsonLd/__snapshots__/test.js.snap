// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotographJsonLd component renders props into values 1`] = `null`;
